MONGOID-5407 Add phantom custom field type example (#5361)
mongodb/mongoid: docs/reference/fields.txt

   Venue.where(location: point) # This uses Point.evolve

.. _phantom-custom-field-types:

Phantom Custom Field Types
``````````````````````````

The custom field type may perform conversions from user-visible attribute
values to the values stored in the database when the user-visible attribute
value type is different from the declared field type. For example, this
can be used to implement a mapping from one enumeration to another, to
have more descriptive values in the application and more compact values stored
in the database:

.. code-block:: ruby

  class ColorMapping
    
    MAPPING = {
      'black' => 0,
      'white' => 1,
    }.freeze
    

>     INVERSE_MAPPING = MAPPING.invert

Author: ```suggestion
    INVERSE_MAPPING = MAPPING.invert.freeze
```

    class << self

>       # Takes any possible object and converts it to how it would be

Author: ```suggestion
      # Takes application-scope value and converts it to how it would be
```

      # stored in the database. Converts invalid values to nil.
      def mongoize(object)
        MAPPING[object]
      end

      # Get the object as it was stored in the database, and instantiate

>       # this custom class from it. Converts invalid values to nil.

Author: ```suggestion
      # value. Converts invalid values to nil.
```
